src/postgres_mcp/database_health/sequence_health_calc.py

```diff
@@ -0,0 +1,152 @@
+
+        if not metrics:
+            return "No sequences found in the database."
+
+        # Sort by remaining values ascending to show most critical first
+        metrics.sort(key=lambda x: x.max_value - x.last_value)
+
+        unhealthy = [m for m in metrics if not m.is_healthy]
+        if not unhealthy:
+            return "All sequences have healthy usage levels."
+
+        result = ["Sequences approaching maximum value:"]
+        for metric in unhealthy:
+            remaining = metric.max_value - metric.last_value
+            result.append(
+                f"Sequence '{metric.schema}.{metric.sequence}' used for {metric.table}.{metric.column} "
+                f"has used {metric.percent_used:.1f}% of available values "
+                f"({metric.last_value:,} of {metric.max_value:,}, {remaining:,} remaining)"
+            )
+        return "\n".join(result)
+
+    async def _get_sequence_metrics(self) -> list[SequenceMetrics]:
+        """Get metrics for sequences in the database."""
+        # First get all sequences used as default values
+        sequences = await self.sql_driver.execute_query("""
+            SELECT
+                n.nspname AS table_schema,
+                c.relname AS table,
+                attname AS column,
+                format_type(a.atttypid, a.atttypmod) AS column_type,
+                pg_get_expr(d.adbin, d.adrelid) AS default_value
+            FROM
+                pg_catalog.pg_attribute a
+            INNER JOIN
+                pg_catalog.pg_class c ON c.oid = a.attrelid
+            INNER JOIN
+                pg_catalog.pg_namespace n ON n.oid = c.relnamespace
+            INNER JOIN
+                pg_catalog.pg_attrdef d ON (a.attrelid, a.attnum) = (d.adrelid, d.adnum)
+            WHERE
+                NOT a.attisdropped
+                AND a.attnum > 0
+                AND pg_get_expr(d.adbin, d.adrelid) LIKE 'nextval%'
+                AND n.nspname NOT LIKE 'pg\\_temp\\_%'
+        """)
+
+        if not sequences:
+            return []
+
+        result_list = [dict(x.cells) for x in sequences]
+
+        # Process each sequence
+        sequence_metrics = []
+        for seq in result_list:
+            # Parse the sequence name from default value
+            schema, sequence = self._parse_sequence_name(seq["default_value"])
+            if not sequence:
+                continue
+
+            # Determine max value based on column type
+            max_value = (
+                2147483647 if seq["column_type"] == "integer" else 9223372036854775807
+            )
+
+            # Get sequence attributes
+            attrs = await SafeSqlDriver.execute_param_query(
+                self.sql_driver,
+                """
+                SELECT
+                    has_sequence_privilege('{}.{}', 'SELECT') AS readable,
+                    last_value
+                FROM {}.{}
+            """,
+                [schema, sequence, schema, sequence],
+            )
```

Comment:
This usage is incorrect. The parameters should only be literal parameters and this functions quotes them.
```suggestion
            attrs = await SafeSqlDriver.execute_param_query(
                self.sql_driver,
                """
                SELECT
                    has_sequence_privilege('{}.{}', 'SELECT') AS readable,
                    last_value
                FROM {}.{}
            """,
                [Identifier(schema, sequence), Identifier(schema, sequence)],
            )
```

But doing it requires changing `param_sql_to_query` in `safe_sql.py`:

```
    @staticmethod
    def param_sql_to_query(query: str, params: list[Any]) -> str:
        """Convert a SQL string to a query string."""
        from psycopg.sql import SQL, Literal, Composable
        
        # Convert each parameter to a Literal only if it's not already a Composable
        sql_params = [p if isinstance(p, Composable) else Literal(p) for p in params]
        
        return SafeSqlDriver.sql_to_query(
            SQL(query).format(*sql_params)  # type: ignore
        )
```